verifiable-db/src/row_tree/full_node.rs

```diff
@@ -44,19 +43,13 @@ impl FullNodeCircuit {
         let value = row.value();
         let digest = row.digest(b, &cells_pi);
 
-        // Check multiplier_vd and row_id_multiplier are the same as children proofs.
+        // Check multiplier_vd and multiplier_counter are the same as children proofs.
         // assert multiplier_vd == p1.multiplier_vd == p2.multiplier_vd
         b.connect_curve_points(digest.multiplier_vd, min_child.multiplier_digest_target());
         b.connect_curve_points(digest.multiplier_vd, max_child.multiplier_digest_target());
-        // assert row_id_multiplier == p1.row_id_multiplier == p2.row_id_multiplier
-        b.connect_biguint(
-            &digest.row_id_multiplier,
-            &min_child.row_id_multiplier_target(),
-        );
-        b.connect_biguint(
-            &digest.row_id_multiplier,
-            &max_child.row_id_multiplier_target(),
-        );
+        // assert multiplier_counter == p1.multiplier_counter == p2.multiplier_counter
+        b.connect(digest.multiplier_cnt, min_child.multiplier_counter_target());
+        b.connect(digest.multiplier_cnt, max_child.multiplier_counter_target());
```

Comment:
Shouldn't we also check that the individual counter are the same between the children and this node ?

Reply:
The individual counter is not exposed as public input in rows tree nodes, since it's already included in the value digest, to compute [`row_id_individual`](https://www.notion.so/lagrangelabs/Generic-Extraction-incl-structs-10c28d1c65a88072b333eea5e1df8d02?pvs=4#11328d1c65a8808baa96d87903cf981f). This should already be enforced to be the same in MPT circuits, where we ensure that `row_id_individual` for each item being extracted is computed using a counter provably bound to the number of columns specified in the metadata digest; therefore, by checking in MPT circuits that the metadata digest is always the same across all the extracted items, we should implicitly check that all counters employed to compute `row_id_individual`, and so the value digest, are the same for all extracted items.